### ISSUE TITLE
Library: visual glitch on removing item from Crate, cover art stays, rows move up but ~16 above

### ISSUE THREAD

**Issue by mxmilkiib** [mixxxdj/mixxx#12565]
### Bug Description

I'm seeing this thing that on removing an item , there's a kind of glitch, the cover art for that row doesn't disappear, it steys the same one is removed like 16 items up, and other tracks' text all jump up one row, but the cover art stays the same height.

### Version

main

### OS

Arch

**Comment by ronso0** [comment]
Could you please provide a short screencast demonstrating the issue?

**Comment by mxmilkiib** [comment]
Took a few attempts.. finally got a clean one. 6s.
[mixxx_library_glitch-2024-01-15_01.59.19.webm](https://github.com/mixxxdj/mixxx/assets/108225/ccf82d1b-9ac7-41e5-bae9-5b64ecb65010)

**Comment by mxmilkiib** [comment]
Another. Accidendally amended it to the first, 7 secs on, ffmpeg editing keeps cutting it wrong and I'm tired so full clip
[mixxx_library_glitch2-2024-01-15_02.09.07.webm](https://github.com/mixxxdj/mixxx/assets/108225/d3cf6330-6686-4271-bbe0-aa9d1c022626)

**Comment by ronso0** [comment]
It's very hard to spot it.
By which column are the track sorted when you remove a track?

Can you reproduce when sort, then remove  a track without doing anything in between that might affect the track metadata?
(e.g. playing a track chnages the play count and the 'last played' value)

**Comment by mxmilkiib** [comment]
Don't play the videos, scrub them so you control what is being shown. Open on mpv and use . and , to move by frame.

I don't know how to reproduce yet. I'm focusing on curating and editing metadata, but I'll see if I can remember. I clicked on the side panel then tab into the library, that might count for something to help reproduced. Probably because I'm on the current Qt.

**Comment by ronso0** [comment]
First clip:
* two rows are removed: the one selected + the one above
* though play counter is only decreased by one
= something happens with the sorting

Same in the second clip:
* one row removed play counter decreased by one)
* tracks in another region are resorted
* covers < > tracks still identical

Currenty you have tracks sorted by key.
Mixxx supports multi-column sorting, so obviously the previous sort columns is relevant here.
A possible explanantion is you previously sorted by Preview (= shuffle, hidden feature).

Please verify sort order is not touched when you sort by two regular properties (columns), then remove a track.

**Comment by mxmilkiib** [comment]
Oh.. If you delete a track, does Tracks resort everything automatically, without one telling it to do so, that where I'm assuming that that sorting behaviour is most often reactivated by clicking the column header.

**Comment by ronso0** [comment]
No, it shouldn't resort, and it doesn't when you remove tracks, even if one or more of the listed tracks were changed since the initial sorting (changed like any track properties used for sorting was touched).

That's
It seems this only only occurs when one of the last **three** columns used for sorting is **Preview**, even if **no listed track** was changed since the initial sorting.

That's why I asked you to
> Please verify sort order is not touched when you sort by two regular properties (columns), then remove a track.

If you can confirm it only occurs when Preview is involved we'll change the title and know where to look (though no guarantee this can be fixed easily).

**Comment by mxmilkiib** [comment]
I haven't sorted by Preview

**Comment by ronso0** [comment]
Oh, you're right. Preview is irrelevant, it's indeed the **last-used** sort column :|

**Comment by ronso0** [comment]
I'm currently cleaning up my library and this is indeed an annoying issue.
Sort tracks by "last played", play some tracks, remove (purge) a track -> table is resorted by "last played"

Working on a fix, looks good so far. For purging and hiding as well.

**Comment by ronso0** [comment]
#15872
